src/clientRPC/utils.ts

```diff
@@ -65,201 +60,149 @@ function encodeAuthFromPassword(password: string): string {
   return `Basic ${encoded}`;
 }
 
-function readableFromWebSocket(
-  ws: WebSocket,
-  logger: Logger,
-): ReadableStream<Uint8Array> {
-  return new ReadableStream<Uint8Array>({
-    start: (controller) => {
-      logger.info('starting');
-      const messageHandler = (data) => {
-        logger.debug(`message: ${data.toString()}`);
-        ws.pause();
-        const message = data as Buffer;
-        if (message.length === 0) {
-          logger.info('ENDING');
-          ws.removeAllListeners('message');
-          try {
-            controller.close();
-          } catch {
-            // Ignore already closed
-          }
-          return;
-        }
-        controller.enqueue(message);
-      };
-      ws.on('message', messageHandler);
-      ws.once('close', () => {
-        logger.info('closed');
-        ws.removeListener('message', messageHandler);
-        try {
-          controller.close();
-        } catch {
-          // Ignore already closed
-        }
-      });
-      ws.once('error', (e) => {
-        controller.error(e);
-      });
-    },
-    cancel: () => {
-      logger.info('cancelled');
-      ws.close();
-    },
-    pull: () => {
-      logger.debug('resuming');
-      ws.resume();
-    },
-  });
+function detailedToCertChain(
+  cert: DetailedPeerCertificate,
+): Array<Certificate> {
+  const certChain: Array<Certificate> = [];
+  let currentCert = cert;
+  while (true) {
+    certChain.unshift(new x509.X509Certificate(currentCert.raw));
+    if (currentCert === currentCert.issuerCertificate) break;
+    currentCert = currentCert.issuerCertificate;
+  }
+  return certChain;
 }
 
-function writeableFromWebSocket(
-  ws: WebSocket,
-  holdOpen: boolean,
-  logger: Logger,
-): WritableStream<Uint8Array> {
-  return new WritableStream<Uint8Array>({
-    start: (controller) => {
-      logger.info('starting');
-      ws.once('error', (e) => {
-        logger.error(`error: ${e}`);
-        controller.error(e);
-      });
-      ws.once('close', (code, reason) => {
-        logger.info(
-          `ws closing early! with code: ${code} and reason: ${reason.toString()}`,
+/**
+ * Verify the server certificate chain when connecting to it from a client
+ * This is a custom verification intended to verify that the server owned
+ * the relevant NodeId.
+ * It is possible that the server has a new NodeId. In that case we will
+ * verify that the new NodeId is the true descendant of the target NodeId.
+ */
+async function verifyServerCertificateChain(
+  nodeIds: Array<NodeId>,
+  certChain: Array<Certificate>,
+): Promise<NodeId> {
```

Comment:
This looks like it came from the `network` domain. You should use the `grpc` domain instead.

The verification for the client service is actually different. And I believe originally it's under the `client` domain.

Reply:
For example the client service does not require verifying multiple Node IDs.

Reply:
Look for where we are verifying the server certificate chain on the GRPC client. (Which is different from how we do this via the TLS inside the `network` domain).

Reply:
I just doubled checked. This is the `VerifyServerCertificateChain` taken from the GRPC code which is using the method defined in the network utils. 

I'm leaving this as-is.